src/layouts/home-assistant.ts

```diff
@@ -57,14 +61,20 @@ export class HomeAssistantAppEl extends QuickBarMixin(HassElement) {
   protected render() {
     const hass = this.hass;
 
-    return hass && hass.states && hass.config && hass.services
-      ? html`
-          <home-assistant-main
-            .hass=${this.hass}
-            .route=${this._route}
-          ></home-assistant-main>
-        `
-      : html`<ha-init-page .error=${this._error}></ha-init-page>`;
+    if (hass && hass.states && hass.config && hass.services) {
+      removeLaunchScreen();
```

Comment:
We should not do this on every render. `render()` cannot have side effects.

Makes me wonder if we should drop `LitElement` here and extend `ReactiveElement` directly so we can implement `update()` (which is meant to have side effects)

Reply:
LitElement actually allows to implement update() so that should be an easy fix

Reply:
LitElement = Reactive Element with an update implementation being kinda like `render(this.shadowRoot, this.render())`. Looked it up just now, I was spot on :-) https://github.com/lit/lit/blob/main/packages/lit-element/src/lit-element.ts#L136

We shouldn't override `update()` from LitElement, but instead implement it for ReactiveElement.

Reply:
But we still need the render() method to from LitElement to render `<home-assistant-main>`

I think the usual way to do this would be to implement update() like this:
```typescript

  update(changedProps) {
    super.update(changedProps);

    const hass = this.hass;
    if (this._isHassComplete()) {
      removeLaunchScreen();
    }
  }

```